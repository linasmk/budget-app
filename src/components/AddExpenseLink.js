/* ========= Dependencies ============= */
import React from "react";
import { NavLink } from "react-router-dom";
/* ========== Code ========== */
export const AddExpenseLink = () => {
  return (
    <NavLink
      to="/create"
      className="add-expense-link"
      activeClassName="is-active-navlink"
    >
      Add Expense
    </NavLink>
  );
};
export default AddExpenseLink;
