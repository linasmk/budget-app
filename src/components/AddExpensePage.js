import React from "react";
const AddExpensePage = () => <div>This is Add Expense Component</div>;
export default AddExpensePage;
