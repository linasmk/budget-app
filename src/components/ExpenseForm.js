/* ===== App Dependencies ===== */
import React from "react";
import moment from "moment";

import { SingleDatePicker } from "react-dates";

// -----------------------------------------
import { cl } from "../utils/DebugHelpers";
/* ========== Code ========== */

export default class ExpenseForm extends React.Component {
  state = {
    description: this.props.expense ? this.props.expense.description : "",
    note: this.props.expense ? this.props.expense.note : "",
    amount: this.props.expense
      ? (this.props.expense.amount / 100).toString()
      : "",
    createdAt: this.props.expense
      ? moment(this.props.expense.createdAt)
      : moment(),
    calenderFocused: false,
    error: "",
  };

  onDescriptionChange = (e) => {
    const description = e.target.value;

    this.setState(() => ({
      description,
    }));
  };
  onNoteChange = (e) => {
    const note = e.target.value;
    this.setState(() => ({
      note,
    }));
  };
  onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };
  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calenderFocused: focused }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.description || !this.state.amount) {
      this.setState(() => ({
        error: "Description and amount fields should not stay empty!",
      }));
    } else {
      this.setState(() => ({
        error: "",
      }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note,
      });
    }
  };

  render() {
    return (
      <section className="form-wrapper">
        {this.state.error && (
          <p className="expense__form-wrapper--warning">{this.state.error}</p>
        )}

        <form className="expense__form" onSubmit={this.onSubmit}>
          <div className="expense-form__input-wrapper">
            <input
              type="text"
              placeholder="Description"
              autoFocus
              value={this.state.description}
              onChange={this.onDescriptionChange}
              className="expense-form__input h-input-styles"
            />
            <input
              type="text"
              placeholder="Amount"
              value={this.state.amount}
              onChange={this.onAmountChange}
              className="expense-form__input h-input-styles"
            />
          </div>

          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calenderFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
            daySize={33}
          />
          <textarea
            placeholder="Add a note (optional)"
            value={this.state.note}
            onChange={this.onNoteChange}
            rows="5"
            className="expense-form__textarea h-input-styles"
          ></textarea>
          <button className="add-expense-btn">Add Expense</button>
        </form>
      </section>
    );
  }
}
