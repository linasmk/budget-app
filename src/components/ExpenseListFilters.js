/* ========= Dependencies ============= */
import React from "react";
import { connect } from "react-redux";
import { DateRangePicker } from "react-dates";
import selectExpenses from "../selectors/expenses";
import {
  setTextFilter,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate,
} from "../actions/filters";

/* ========== Code ========== */
export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({
      calendarFocused,
    }));
  };
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = (e) => {
    if (e.target.value === "date") {
      this.props.sortByDate();
    } else if (e.target.value === "amount") {
      this.props.sortByAmount();
    }
  };

  render() {
    return (
      <div className="expense-filters">
        <div className="expense-filters__input-wrapper">
          <input
            type="text"
            value={this.props.filters.text}
            onChange={this.onTextChange}
            placeholder="Expense description"
            className="expense-filters__input--text h-input-styles"
          />
          <select
            value={this.props.filters.sortBy}
            onChange={this.onSortChange}
            className="expense-filters__input--select h-input-styles"
          >
            <option value="date">Date</option>
            <option value="amount">Amount</option>
          </select>
        </div>

        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          startDateId={"dwjkhqkehwqjkeq"}
          endDateId={"cxzvcxbzbczxbz"}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
          daySize={33}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};
const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
