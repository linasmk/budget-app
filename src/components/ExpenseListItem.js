/* ========= App Dependencies ============= */
import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

/* ========= Code ============= */

const ExpenseListItem = ({ id, description, note, amount, createdAt }) => (
  <div className="expens-item">
    <h3 className="expense-item__description">
      <Link className="expens-item__link" to={`/edit/${id}`}>
        {description}
      </Link>
    </h3>
    <p className="expense-item__date">
      {moment(createdAt).format("MMMM Do, YYYY. [At:] LT")}
    </p>
    <p className="expense-item__amount">
      {"â‚¬" + numeral(amount / 100).format("0,0.00")}
    </p>
    <p className="expense-item__note">{note}</p>
  </div>
);

export default ExpenseListItem;
