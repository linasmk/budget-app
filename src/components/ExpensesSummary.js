/* ========= App Dependencies ============= */
import React from "react";
import numeral from "numeral";
import { connect } from "react-redux";
/* ========== Redux ========== */
import getVisibleExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

/* ========== Components ========== */
/* ========= Code ============= */
export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
  const expenseCountSpelling = expenseCount === 1 ? "expense" : "expenses";
  return (
    <div>
      <p>
        {expenseCount} {expenseCountSpelling} visible given by your search.
      </p>
      <p>Totalling {"â‚¬" + numeral(expenseTotal / 100).format("0,0.00")} .</p>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expenseTotal: selectExpensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
