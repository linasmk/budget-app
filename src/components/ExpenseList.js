import React from "react";
import { connect } from "react-redux";
/* ========== Redux ========== */
import getVisibleExpenses from "../selectors/expenses";
/* ========== Components ========== */
import ExpenseListItem from "./ExpenseListItem";

/* ========== Code ========== */

const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    {props.expenses.map((expense) => {
      return <ExpenseListItem key={expense.id} {...expense} />;
    })}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters),
  };
};

export default connect(mapStateToProps)(ExpenseList);
