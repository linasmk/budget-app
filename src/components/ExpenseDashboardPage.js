/* ========= Dependencies ============= */
import React from "react";
/* ========== Components ========== */

import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import AddExpenseLink from "./AddExpenseLink";

/* ========== Code ========== */

const ExpenseDashboardPage = () => (
  <div className="expense-dashboard">
    <article className="expense-dashboard__content">
      <ExpenseListFilters />
      <AddExpenseLink />
      <ExpenseList />
    </article>
  </div>
);

export default ExpenseDashboardPage;
