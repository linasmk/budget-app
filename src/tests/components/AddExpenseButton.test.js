/* ========= App Dependencies ============= */
import React from "react";
import { shallow } from "enzyme";
/* ========== Components ========== */
import { AddExpenseLink } from "../../components/AddExpenseLink";
/* ========== Code ========== */
it("renders AddExpenseLink correctly", () => {
  const wrapper = shallow(<AddExpenseLink />);
  expect(wrapper).toMatchSnapshot();
});
