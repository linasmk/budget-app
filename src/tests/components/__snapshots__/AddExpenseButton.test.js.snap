// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddExpenseLink correctly 1`] = `
<NavLink
  activeClassName="is-active-navlink"
  className="add-expense-link"
  to="/create"
>
  Add Expense
</NavLink>
`;
